smarts/core/chassis.py

```diff
@@ -151,10 +159,14 @@ def __init__(
         bullet_client.setCollisionFilterGroupMask(
             self._bullet_body._bullet_id, -1, 0x0, 0x0
         )
+        self._pose = None
         self.control(pose, speed)
         self._client = bullet_client
 
-    def control(self, pose: Pose, speed: float):
+    def control(self, pose: Pose, speed: float, dt: float = 0):
+        if self._pose:
+            self._last_heading = self._pose.heading
+        self._last_dt = dt
```

Comment:
I somewhat worry about holding onto the last timestep opens the chance of de-synchronization between the simulator state and vehicle state.

Reply:
Yeah, I didn't like this either (for a different reason).  I'll try to improve this...

Reply:
... actually, after looking at it again, I don't see any obvious ways to easily clean this up.  Do you have any suggestions?

Reply:
I am also a bit stumped other than getting the value from a simulation time utility which opens up its own problems.

Reply:
Yeah, I agree.  I think I'm going to leave it as is then.